libs/core/src/lib/effect.spec.ts

```diff
@@ -1,3 +1,121 @@
-it('', () => {
-  expect(true).toBe(true);
+import { of, timer } from 'rxjs';
+import { Effect } from './effect';
+import { setup } from './test-helper';
+
+it('spawns and tears down an empty effect', () => {
+  const effect: Effect<{}> = () => of();
+  setup({ value: {}, effect }, ({ expectObservable, effect$, storeEvent$ }) => {
+    expectObservable(effect$).toBe('|');
+    expectObservable(storeEvent$).toBe('(ab)', {
+      a: { event: 'spawn', name: 'root', type: 'effect' },
+      b: { event: 'teardown', name: 'root', type: 'effect' },
+    });
```

Comment:
The idea w/ this style of tests is to verify what would otherwise be an "implementation detail", but is actually a contract I want to enforce, since I intend this `storeEvent$` stream to be part of our "public API", breaking changes should require a major version bump, and these tests should catch any issues like running effects more or less often than expected, or in different orders, even if the effect otherwise produces the same values.

I don't use the enums but rather add raw strings, because again its public API, and changes to the values in the enum should be considered breaking, so its undesirable to use an enum in the test where the test could easily be updated erroneously by using a refactoring tool. I'd like this test to fail, even if someone applies a "refactor", because its not really a refactor, its part of the public API & therefore it'd just be a behavior change not a refactor if it were to change.